### PR TITLE
Rename parse_redirect to try_parse_redirect

### LINKED ISSUE #73: Rename `parse_redirect` to `try_parse_redirect`
## Description

In the current implementation, `parse_redirect` returns a `bool`, but the prefix `parse_` is typically associated with functions returning a `t_ast *`.
There is room for improvement by changing the name of the function to `try_parse_redirect` to explicitly indicate that it returns a boolean value.

### PR DESCRIPTION
**Pull request by rask24**
fix #73

## Summary by CodeRabbit

- **New Features**
	- Introduced a new parsing strategy for redirect tokens, enhancing the handling of command parsing.
- **Bug Fixes**
	- Updated function names to reflect improved parsing logic, ensuring more accurate processing of redirect tokens.
- **Documentation**
	- Updated comments and headers to reflect recent changes and modifications.

### CODE REVIEW

File: src/parser/parser_internal.h [hunk 1]
```diff
@@ -24,7 +24,8 @@ t_ast	*parse_simple_command(t_token_list **cur_token);
 t_ast	*parse_pipeline(t_token_list **cur_token);
 t_ast	*parse_command(t_token_list **cur_token);
 t_ast	*parse_list(t_token_list **cur_token);
-bool	parse_redirect(t_ast *node, t_token_list **cur_token);
+
+bool	try_parse_redirect(t_ast *node, t_token_list **cur_token);
```
[the remark this answers is not in the document]
Comment (rask24) [reply src/parser/parser_internal.h:28]: I think Makefile is already correct, but it is need to change the comment in `src/parser/try_parse_redirect.c`.
I've fixed! THX!

**Review by alex216 — COMMENTED** [review]
how about `bool parse_cmd_arg()`? should that also be `try_parse_cmd_arg()`?
@rask24

**Comment by rask24** [comment]
@alex216 Exactly.
I've renamed 👍🏼

**Review by alex216 — APPROVED** [review]
@rask24 ok！

**Pull request merged**